azure-pipelines.yml

```
@@ -38,9 +38,21 @@ jobs:
   - script: python -m flake8
     displayName: 'Run linter'
 
-  - script: BOT_TOKEN=foobar python -m pytest tests
+  - script: BOT_TOKEN=foobar python -m pytest --junitxml=junit.xml --cov=bot --cov-branch --cov-report=term --cov-report=xml tests
     displayName: Run tests
 
+  - task: PublishCodeCoverageResults@1
+    displayName: 'Publish Coverage Results'
+    condition: succeededOrFailed()
+    inputs:
+      codeCoverageTool: Cobertura
+      summaryFileLocation: coverage.xml
+
+  - task: PublishTestResults@2
+    inputs:
+      testResultsFiles: junit.xml
```

I don't know if pytest returns 1 when it fails. If it does then ensuring the results are published regardless would be a good idea.
```suggestion
      condition: succeededOrFailed()
      testResultsFiles: junit.xml
```

---

Added